### PR TITLE
Add override for message that accepts a location.

### PR DESCRIPTION
**Pull request by ibuclaw**
And added backticks to the messages so they _could_ be syntax highlighted, but it's probably better to add in a prefix to visually separate the **bold** location and highlighted names before doing that.

### CODE REVIEW

**Comment by ibuclaw** [comment]
Passing a location is useful for gdc, whose diagnostic backend also prints the source line and uses a caret for underlining.

**Review by wilzbach — APPROVED** [review]
Yeah I'm looking forward to the day when DMD will detect that it's run in a TTY and displays more verbose errors messages with the the offending line printed like GCC does ...

File: src/dmd/dsymbolsem.d [hunk 1]
```diff
@@ -654,9 +654,8 @@ private extern(C++) final class DsymbolSemanticVisitor : Visitor
             {
                 if (global.params.vfield && dsym.storage_class & (STC.const_ | STC.immutable_) && dsym._init && !dsym._init.isVoidInitializer())
                 {
-                    const(char)* p = dsym.loc.toChars();
                     const(char)* s = (dsym.storage_class & STC.immutable_) ? "immutable" : "const";
-                    message("%s: %s.%s is %s field", p ? p : "", ad.toPrettyChars(), dsym.toChars(), s);
+                    message(dsym.loc, "`%s.%s` is `%s` field", ad.toPrettyChars(), dsym.toChars(), s);
```
Comment (wilzbach) [inline src/dmd/dsymbolsem.d]: You need to update some tests for this e.g.

```
==============================
Test failed: 
expected:
----
compilable/sw_transition_field.d(15): sw_transition_field.S1.ix is immutable field
compilable/sw_transition_field.d(16): sw_transition_field.S1.cx is const field
compilable/sw_transition_field.d(21): sw_transition_field.S2!(immutable(int)).S2.f is immutable field
compilable/sw_transition_field.d(21): sw_transition_field.S2!(const(int)).S2.f is const field
----
actual:
----
compilable/sw_transition_field.d(15): `sw_transition_field.S1.ix` is `immutable` field
compilable/sw_transition_field.d(16): `sw_transition_field.S1.cx` is `const` field
compilable/sw_transition_field.d(21): `sw_transition_field.S2!(immutable(int)).S2.f` is `immutable` field
compilable/sw_transition_field.d(21): `sw_transition_field.S2!(const(int)).S2.f` is `const` field
----
```

Comment (ibuclaw) [reply src/dmd/dsymbolsem.d]: Right, always let the CI find this out so I can fix up later. ;-)

Should be fine now.

**Review by marler8997 — APPROVED** [review]

**Comment by ibuclaw** [comment]
@wilzbach all fine here?

**Review by JinShil — APPROVED** [review]

**Review by wilzbach — APPROVED** [review]
Thanks!

**Pull request merged**